operators/pkg/controller/common/service.go

```diff
@@ -5,28 +5,34 @@
 package common
 
 import (
-	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 )
 
-const (
-	// DefaultServiceType is used when the stack spec is empty or not valid.
-	DefaultServiceType = corev1.ServiceTypeClusterIP
-)
+// SetServiceDefaults updates the service with the provided defaults if they are not already set.
+func SetServiceDefaults(
```

Comment:
I had to mind to set this kind of stuff in `common/overrides` (for env, labels, services, volumes, volume mounts, etc.), but I'm fine if you think it's better located here.